airbyte-integrations/connectors/destination-postgres/src/test-integration/java/io/airbyte/integrations/destination/postgres/typing_deduping/PostgresSqlGeneratorIntegrationTest.java

```diff
@@ -0,0 +1,154 @@
+import io.airbyte.cdk.integrations.destination.jdbc.TableDefinition;
+import io.airbyte.cdk.integrations.destination.jdbc.typing_deduping.JdbcDestinationHandler;
+import io.airbyte.cdk.integrations.destination.jdbc.typing_deduping.JdbcSqlGenerator;
+import io.airbyte.cdk.integrations.standardtest.destination.typing_deduping.JdbcSqlGeneratorIntegrationTest;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.integrations.base.destination.typing_deduping.DestinationHandler;
+import io.airbyte.integrations.base.destination.typing_deduping.Sql;
+import io.airbyte.integrations.destination.postgres.PostgresDestination;
+import io.airbyte.integrations.destination.postgres.PostgresSQLNameTransformer;
+import io.airbyte.integrations.destination.postgres.PostgresTestDatabase;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Optional;
+import javax.sql.DataSource;
+import org.jooq.DataType;
+import org.jooq.Field;
+import org.jooq.SQLDialect;
+import org.jooq.impl.DSL;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+public class PostgresSqlGeneratorIntegrationTest extends JdbcSqlGeneratorIntegrationTest {
+
+  private static PostgresTestDatabase testContainer;
+  private static String databaseName;
+  private static JdbcDatabase database;
+
+  /**
+   * See
+   * {@link io.airbyte.integrations.destination.redshift.typing_deduping.RedshiftSqlGeneratorIntegrationTest.RedshiftSourceOperations}.
+   * Copied here to avoid weird dependencies.
+   */
+  public static class PostgresSourceOperations extends JdbcSourceOperations {
+
+    @Override
+    public void copyToJsonField(final ResultSet resultSet, final int colIndex, final ObjectNode json) throws SQLException {
+      final String columnName = resultSet.getMetaData().getColumnName(colIndex);
+      final String columnTypeName = resultSet.getMetaData().getColumnTypeName(colIndex).toLowerCase();
+
+      switch (columnTypeName) {
+        // JSONB has no equivalent in JDBCType
+        case "jsonb" -> json.set(columnName, Jsons.deserializeExact(resultSet.getString(colIndex)));
+        // For some reason, the driver maps these to their timezoneless equivalents (TIME and TIMESTAMP)
+        case "timetz" -> putTimeWithTimezone(json, columnName, resultSet, colIndex);
+        case "timestamptz" -> putTimestampWithTimezone(json, columnName, resultSet, colIndex);
+        default -> super.copyToJsonField(resultSet, colIndex, json);
+      }
+    }
+
+  }
+
+  @BeforeAll
+  public static void setupPostgres() {
+    testContainer = PostgresTestDatabase.in(PostgresTestDatabase.BaseImage.POSTGRES_13);
+    final JsonNode config = testContainer.configBuilder()
+        .with("schema", "public")
+        .withDatabase()
+        .withResolvedHostAndPort()
+        .withCredentials()
+        .withoutSsl()
+        .build();
+
+    databaseName = config.get(JdbcUtils.DATABASE_KEY).asText();
```

Comment:
sources and prev DATs were written such that same DB is not resued across tests and isolated with a suffix with `@BeforeEach`